test/e2e/node_feature_discovery_test.go

```diff
@@ -828,6 +828,23 @@ core:
 					By("Verfiying node status capacity from NodeFeatureRules #4")
 					Expect(waitForCapacity(ctx, f.ClientSet, expectedCapacity, nodes)).NotTo(HaveOccurred())
 
+					By("Creating NodeFeatureRules #5")
+					Expect(testutils.CreateNodeFeatureRulesFromFile(ctx, nfdClient, "nodefeaturerule-5.yaml")).NotTo(HaveOccurred())
+
+					By("Verifying node labels from NodeFeatureRules #5")
+					expectedAnnotations := map[string]k8sAnnotations{
+						"*": {
+							nfdv1alpha1.FeatureLabelNs + "defaul-ns-annotation":   "foo",
+							nfdv1alpha1.FeatureLabelNs + "defaul-ns-annotation-2": "bar",
+							"vendor.example/feature":                              "baz",
+						},
+					}
+					Expect(checkForNodeAnnotations(ctx,
+						f.ClientSet,
+						expectedAnnotations,
+						nodes,
+					)).NotTo(HaveOccurred())
```

Comment:
Could you add a step to delete the object and verify that annotations were removed?